pkg/pillar/pubsub/pubsub.go

```diff
@@ -48,7 +47,12 @@ type SubscriptionOptions struct {
 //     case change := <- s1.C:
 //         s1.ProcessChange(change, ctx)
 //  }
-type SubHandler func(ctx interface{}, key string, status interface{})
+type SubCDHandler func(ctx interface{}, key string, status interface{})
```

Comment:
The only thing I can find to suggest here, is that `SubCDHandler` is a bad name. My first reaction when I saw it was to be unsure if it referred somehow to compact disks, or to continuous delivery. "Create Delete" didn't enter my mind.

Either we should rename it `SubCreateModifyHandler`, or (what I like better, but this is personal style), to have three distinct handlers:

```go
type SubCreateHandler func(ctx interface{}, key string, status interface{})
type SubDeleteHandler func(ctx interface{}, key string, status interface{})
type SubModifyHandler func(ctx interface{}, key string, status interface{}, oldStatus interface{})
```

Sure, `SubCreateHandler` and `SubDeleteHandler` have the same signature, but it doesn't hurt, and it is clear and easy to understand.

Reply:
I'll create two separate names for the CD types.

Reply:
Fixed